### ISSUE TITLE
DEL detected in germline mode is not detected in non-germline mode

### ISSUE THREAD

**Issue by waltergallegog** [fritzsedlazeck/Sniffles#399]
Hello,
I have run Sniffles2 in `non-germilne` mode, but it is missing a deletion I was expecting, because it is given by other SV callers (cuteSV, SAVANA, nanomonSV) and because it is present from visual inspection with IGV.

I ran Sniffles2 again, in the default (germline) mode, and this time the deletion is present.
From my understanding, the  `non-germilne` should output both germilne and non-germline variants, as per https://github.com/fritzsedlazeck/Sniffles/issues/287#issuecomment-1033673946
> The germline variants will still be output, but have a heterozygous / homozygous variant genotype (0/1 or 1/1) assigned.

Is the behavior I'm seeing expected?

For some context, I'm running sniffles2 on a tumor/control pair, and then merging them to get the genotyped variants, with a similar workflow as described in https://github.com/fritzsedlazeck/Sniffles/issues/353. For the tumor sample, I'm interested in all variants (either with low high support).
 
Is it correct if I run the tumor sample in both `germline` and `non-germline` modes and then merge them with the control?
Then I could get the somatic variants as those present in the germline and non-germline tumor columns, but not present in the control.

Some details about the problematic deletion in case they are helpful:
When detected by Sniffles2 in default mode it is marked as precise, with quality 60 and its genotype is 0/1 (14:11). Here are the INFO and SAMPLE fields:
```
PRECISE;SVTYPE=DEL;SVLEN=-7669;END=14050579;SUPPORT=11;COVERAGE=24,21,28,26,28;STRAND=+-;AF=0.440;STDEV_LEN=3.109;STDEV_POS=0.000	GT:GQ:DR:DV	0/1:60:14:11
```
 
From IGV I can see the supporting reads with the deletion:
![non-germ-missing-del](https://github.com/fritzsedlazeck/Sniffles/assets/5270762/dcfb446a-a22b-44c9-bcf6-43f3a7273c31)

From the run in `non-germline` mode, I don't see any other variant in the same position or that is in overlap, so I guess is not an issue of a non-germline variant obscuring the germilne one. 

Please let me know if more details are required.

Thanks and BR
Walter

**Comment by waltergallegog** [comment]
As additional info, I have merged the snfs from the control and tumor samples.
We actually have two tumor samples, one taken from the patient (known as tumor0) and one after 6 months of in-lab growth (known as tumor1).

Here is the result from the merge:
| Sample name | GT:GQ:DR:DV:ID |
|---------------------|-----------------------------------|
| Control             | 0/0:0:22:0:NULL                   |
| Tumor0 germline     | 0/1:60:14:11:Sniffles2.DEL.AC9S14 |
| Tumor0 non germline | 0/0:0:24:0:NULL                   |
| Tumor1 germline     | 0/1:14:9:4:Sniffles2.DEL.AF5S14   |
| Tumor1 non germline | 0/1:14:9:4:Sniffles2.DEL.6CDS14   |

Interestingly, for Tumor0 non germline mode, even with genotyping the variant is not found.
Tumor1 has no problem as it is found in both germline and non germline modes.

The reason we are focusing on this DEL is because it is given as somatic with very high confidence by both nanomonsv and savana, and with cutesv we have found it in tumor0 and tumor1 but not in the control. We have also found the deletion with SV genotyping in short reads (illumina)  and we can see it from visual inspection in the long reads (ONT) with IGV.

**Comment by fritzsedlazeck** [comment]
Hi 
yeah this is confusing... the non-germline mode currently is across the entire frequency , but is much more conservative. Thus what we suggest is to run both modes (default and non-germline). 

We are soon to release a version that will have the two modes better split up and thus makes this easier. 

hope that helps and thanks for reaching out 
Fritz

**Comment by waltergallegog** [comment]
Thanks for the feedback.
I will keep using both modes as you suggested.
BR
Walter

**Comment by waltergallegog** [comment]
Sorry for reopening this thread but it seems the most appropriate place.

I have a deletion which is present only in one of the samples (tumor1) and detected only in germline mode.
It is detected with genotype 0/0  (**0/0:11:15:3**) and marked with filter **GT**.

I would expect this deletion to be detected by the non-germline mode, in which case, if I'm not mistaken, it would be marked as PASS, as the GT filter does not apply for this mode. But  as you explained, the non-germline mode is more conservative.

After the merge, the deletion is not present. I suppose it is because it has been detected only in one sample (tumor1 germile) but it has not passed the quality filter as it was marked as GT.

What would you suggest to do in this case? For my use case in which I'm interested in variants in the tumor samples with both high and low support, do you think is a good idea to just disable the GT filter for germline mode as well? Is there an easy way to do it?

Thanks again,
Walter.

**Comment by fritzsedlazeck** [comment]
@lfpaulin can you check the frequency gap is closed?
